scripts/generate-op-funcs.ts

```diff
@@ -0,0 +1,210 @@
+    if (hasFlags(argType, AlgoTsType.Uint64)) yield 'uint64'
+    if (hasFlags(argType, AlgoTsType.Bytes)) yield 'bytes'
+    if (hasFlags(argType, AlgoTsType.String)) yield 'string'
+    if (hasFlags(argType, AlgoTsType.Boolean)) yield 'boolean'
+    if (hasFlags(argType, AlgoTsType.BigUint)) yield 'biguint'
+    if (hasFlags(argType, AlgoTsType.Void)) yield 'void'
+    if (hasFlags(argType, AlgoTsType.Enum)) {
+      for (const enumDef of module.enums.filter((a) => hasFlags(argType, a.typeFlag))) {
+        yield enumDef.tsName
+      }
+    }
+  }
+  function* emitArg(arg: OpArg) {
+    yield arg.name
+    if (arg.optional) {
+      yield '?'
+    }
+    yield ':'
+    yield Array.from(emitArgType(arg.type)).join(' | ')
+    yield ','
+  }
+
+  function combineReturnTypes(op: OpOverloadedFunction): string {
+    return op.signatures
+      .map((s) => Array.from(emitReturnTypes(s.returnTypes)).join(''))
+      .filter(distinct())
+      .join(' | ')
+  }
+
+  yield* emitHeader()
+
+  yield* emitEnums()
+
+  for (const item of module.items) {
+    if (item.type === 'op-function') {
+      yield* emitDoc(item.docs, item.minAvmVersion)
+      yield `export function ${camelCase(item.name)}(`
+      for (const arg of item.immediateArgs) {
+        yield* emitArg(arg)
+      }
+      for (const arg of item.stackArgs) {
+        yield* emitArg(arg)
+      }
+      yield `): `
+      yield* emitReturnTypes(item.returnTypes)
+      yield* noImplementation()
+    } else if (item.type === 'op-overloaded-function') {
+      for (const signature of item.signatures) {
+        yield* emitDoc(signature.docs)
+        yield `export function ${camelCase(item.name)}(`
+        for (const arg of signature.immediateArgs) {
+          yield* emitArg(arg)
+        }
+        for (const arg of signature.stackArgs) {
+          yield* emitArg(arg)
+        }
+        yield `): `
+        yield* emitReturnTypes(signature.returnTypes)
+        yield '\n'
+      }
+      yield `export function ${camelCase(item.name)}(`
+      const args = item.signatures
+        .map((s) => s.immediateArgs.concat(s.stackArgs))
+        .reduce((acc, cur) =>
+          new Array(Math.max(acc.length, cur.length)).fill(null).map((_, i): OpArg => {
+            const left = acc[i]
+            const right = cur[i]
+            return left
+              ? right
+                ? {
+                    name: left.name,
+                    type: left.type | right.type,
+                  }
+                : {
+                    ...left,
+                    optional: true,
+                  }
+              : {
+                  ...right,
+                  optional: true,
+                }
+          }),
+        )
+      for (const arg of args) {
+        yield* emitArg(arg)
+      }
+      yield '): '
+      yield combineReturnTypes(item)
+      yield* noImplementation()
+    } else {
+      yield* emitDoc(item.docs)
+
+      yield `export const ${pascalCase(item.name)} = {`
```

Comment:
I think we should not call `pascalCase` here. It turns `GITxn` to `GiTxn` which I believe is not desirable.